### PR TITLE
Smb hashmap/v6

### PR DESCRIPTION
**Pull request by victorjulien**
#11786 with all hashmaps turned to lru

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23175

File: rust/src/smb/smb.rs [hunk 1]
```diff
@@ -2401,10 +2448,71 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
...
+                if v > 0 {
+                    SMB_CFG_MAX_FRAG_CACHE_SIZE = v;
+                } else {
+                    SCLogError!("Invalid max-dcerpc-frag-cache-size value");
+                }
+            } else {
+                SCLogError!("Invalid max-dcerpc-frag-cache-size value");
+            }
+        }
+        let retval = conf_get("app-layer.protocols.smb.max-ssn2vec-cache-size");
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:2499]: this is a bit of a multi purpose store for mapping ssn id to various things, hard to come up with a good name

Comment (jasonish) [reply rust/src/smb/smb.rs:2499]: Also seems a bit of a low level tuneable. I wonder if such items should be left out of our default configuration file and well documented in an SMB chapter of the user guide.

Comment (victorjulien) [reply rust/src/smb/smb.rs:2499]: yeah probably, but still need a reasonable name :)

**Comment by victorjulien** [comment]
replaced by #12036

**Pull request closed without merging**